staging/src/k8s.io/apiserver/pkg/server/lifecycle_signals.go

```diff
@@ -143,6 +147,7 @@ func newLifecycleSignals() lifecycleSignals {
 	return lifecycleSignals{
 		ShutdownInitiated:          newNamedChannelWrapper("ShutdownInitiated"),
 		AfterShutdownDelayDuration: newNamedChannelWrapper("AfterShutdownDelayDuration"),
+		PreShutdownHooksStopped:    newNamedChannelWrapper("PreShutdownHooksStopped"),
```

Comment:
don't we have a test that should fail now?

https://github.com/kubernetes/kubernetes/blob/f727b5af34dac134aaa5bb5a8f02f78ed6e5291c/staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go#L227-L232

Reply:
no, i am not wrapping `PreShutdownHooksStopped` in the test yet, preshutdown hook is still buggy, https://github.com/kubernetes/kubernetes/pull/110026 fixes the bug and the test wraps the hook as well.